client/lib/query-manager/activity/index.js

```diff
@@ -0,0 +1,78 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { overEvery, get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import QueryManager from 'lib/query-manager';
+
+/**
+ * ActivityQueryManager manages Activity which can be queried
+ */
+export default class ActivityQueryManager extends QueryManager {
+	constructor( data, options ) {
+		super( data, {
+			itemKey: 'ts_utc',
+			...options,
+		} );
+	}
+
+	/**
+	 * Sort descending order, defaulting to end of list if unknown.
+	 *
+	 * @param  {Object} query Query object (Unused).
+	 * @param  {Object} itemA First item
+	 * @param  {Object} itemB Second item
+	 * @return {Number}       0 if equal, less than 0 if itemA is first,
+	 *                        greater than 0 if itemB is first.
+	 */
+	compare( query, { ts_utc: tsA }, { ts_utc: tsB } ) {
+		return tsB - tsA;
+	}
+
+	/**
+	 * Returns true if the item matches the given query, or false otherwise.
+	 *
+	 * @param  {Object}  query Query object
+	 * @param  {Object}  item  Item to consider
+	 * @return {Boolean}       Whether item matches query
+	 */
+	matches = ActivityQueryManager.matches;
```

Comment:
Thanks for the feedback!

This may be an over-optimization, but with this I wanted to save creating the `matches` method on all the `ActivityQueryManager` instances. It's really a pure function that can be defined once statically on the class.

`QueryManager` calls the instance method `matches` in two places:

https://github.com/Automattic/wp-calypso/blob/0d4b14607bf9b6c147b88b4301e85648668d07bd/client/lib/query-manager/index.js#L329-L346

I see the following options:
* Replace the docblock with an improved description that the instance method is just a reference to the (pure) class method.
* Just drop the optimization which probably provides trivial benefit and a comprehension penalty.
* Take a deeper look at `QueryManager` and its subclasses to see whether `matches` could/should be static. A cursory look indicates that it probably could be.

I'm inclined to either replace the docblock or just drop the static method and declare it as an instance method, whatever you think is best.

A semi-related question: When calling these static methods, I find `ActivityQueryManger.f()` to be the most readable syntax, but is slightly more fragile than `this.constructor.f()`. Do you have thoughts on that?